lib/active_merchant/billing/gateways/flo2cash_rest.rb

```diff
@@ -0,0 +1,595 @@
+          branchAddress2: options[:bank_address2],
+          account: {
+            name: options[:account_name],
+            number: options[:account_number]
+          }
+        }
+      end
+
+      # Add channel
+      #
+      # ==== Options
+      #
+      # * <tt>channel</tt> -- One of { web | vt | api | batch | recurring | ivr | mobile }
+      #
+      def add_channel(post, options)
+        post[:channel] = options[:channel] || 'web'
+      end
+
+      # Add initial payment
+      #
+      # ==== Options
+      #
+      # * <tt>initial_date</tt> -- Initial payment date
+      # * <tt>intital_amount</tt> -- Amount to bill
+      #
+      def add_initial_payment(post, options)
+        post[:initialPayment] = {
+          'date' => options[:initial_date] || options[:start_date] || Date.today.at_beginning_of_month.next_month,
+          'amount' => amount(options[:initial_amount]) || amount(options[:amount])
+        }
+      end
+
+      #
+      # instalmentFailOption
+      # "none | add-to-next | add-to-last | add-at-end"
+      #
+      def add_invoice(post, money, options)
+        add_amount(post, money)
+        post[:currency] = options[:currency] || currency(money)
+        if options.has_key?('total_amount')
+          post[:totalAmount] = options[:total_amount]
+          post[:instalmentFailOption] = options[:instalment_fail_option] || 'none'
+        end
+      end
+
+      def add_amount(post, money)
+        post[:amount] = amount(money)
+      end
+
+      # Add frequency
+      #
+      # ==== Options
+      #
+      # * <tt>frequency</tt>: one of { daily | weekly | fortnightly | 4-weekly | 8-weekly | 12-weekly |
+      # monthly | 2-monthly | 3-monthly | 6-monthly | 12-monthly }
+      #
+      def add_frequency(post, options)
+        post[:frequency] = options[:frequency]
+      end
+
+      def add_merchant(post, options)
+        post[:merchant]  = {
+          'id' => options[:merchant_id] || @merchant_id
+          # 'subAccount' => options[:sub_account_id] || @sub_account_id
+        }
+      end
+
+      # Add payment method
+      #
+      # ==== Options
+      #
+      # * <tt>payment_method_type</tt>: token | iframe | card
+      # * <tt>token</tt>: token | iframe | card
+      #
+      def add_payment_method(post, options, value)
+        if options[:payment_method_type] == 'iframe'
+          post[:paymentMethod] = {
+            type: options[:payment_method_type],
+            iframe: { 'value': value }
+          }
+
+        else
+          post[:paymentMethod] = {
+            type: options[:payment_method_type] || 'token',
+            token: { 'value': value }
+          }
+        end
+      end
+
+      def add_references(post, options)
+        post[:reference1] = options[:reference_1]
+        post[:reference2] = options[:reference_2]
+      end
+
+      def add_debit_card_references(post, options)
+        post[:statementReference1] = options[:statementReference1]
```

Comment:
Same comment as earlier and also to keep it consistent with a more ruby style, rename the `options` keys to `:statement_reference_1` etc ...